src/core/lib/security/authorization/rbac_translator.cc

```diff
@@ -328,24 +335,140 @@ absl::StatusOr<std::map<std::string, Rbac::Policy>> ParseRulesArray(
+    policies[policy_name] = std::move(policy_or.value());
   }
   return std::move(policies);
 }
 
 absl::StatusOr<Rbac> ParseDenyRulesArray(const Json& json,
                                          absl::string_view name) {
-  auto policies_or = ParseRulesArray(json, name);
+  auto policies_or = ParseRulesArray(json);
   if (!policies_or.ok()) return policies_or.status();
-  return Rbac(Rbac::Action::kDeny, std::move(policies_or.value()));
+  return Rbac(Rbac::Action::kDeny, std::move(policies_or.value()), name);
 }
 
 absl::StatusOr<Rbac> ParseAllowRulesArray(const Json& json,
                                           absl::string_view name) {
-  auto policies_or = ParseRulesArray(json, name);
+  auto policies_or = ParseRulesArray(json);
   if (!policies_or.ok()) return policies_or.status();
-  return Rbac(Rbac::Action::kAllow, std::move(policies_or.value()));
+  return Rbac(Rbac::Action::kAllow, std::move(policies_or.value()), name);
+}
+
+absl::StatusOr<std::unique_ptr<experimental::AuditLoggerFactory::Config>>
+ParseAuditLogger(const Json& json, size_t pos) {
+  if (json.type() != Json::Type::kObject) {
+    return absl::InvalidArgumentError(
+        absl::StrFormat("\"audit_loggers[%d]\" is not an object.", pos));
+  }
+  bool is_optional = false;
+  auto it = json.object().find("is_optional");
+  if (it != json.object().end()) {
+    switch (it->second.type()) {
+      case Json::Type::kTrue:
+        is_optional = true;
+        break;
+      case Json::Type::kFalse:
+        break;
+      default:
+        return absl::InvalidArgumentError(absl::StrFormat(
+            "\"audit_loggers[%d].is_optional\" is not a boolean.", pos));
+    }
+  }
+  it = json.object().find("name");
+  if (it == json.object().end()) {
+    return absl::InvalidArgumentError(
+        absl::StrFormat("\"audit_loggers[%d].name\" is required.", pos));
+  }
+  if (it->second.type() != Json::Type::kString) {
+    return absl::InvalidArgumentError(
+        absl::StrFormat("\"audit_loggers[%d].name\" is not a string.", pos));
+  }
+  absl::string_view name = it->second.string();
+  Json config;
+  it = json.object().find("config");
+  if (it != json.object().end()) {
+    if (it->second.type() != Json::Type::kObject) {
+      return absl::InvalidArgumentError(absl::StrFormat(
+          "\"audit_loggers[%d].config\" is not an object.", pos));
+    }
+    config = it->second;
+  } else {
+    return absl::InvalidArgumentError(
+        absl::StrFormat("\"audit_loggers[%d].config\" is required.", pos));
+  }
+  if (!AuditLoggerRegistry::FactoryExists(name)) {
+    if (is_optional) {
+      return nullptr;
+    }
+    return absl::InvalidArgumentError(absl::StrFormat(
+        "\"audit_loggers[%d].name\" %s is not supported.", pos, name));
+  }
+  auto result = AuditLoggerRegistry::ParseConfig(name, config);
+  if (!result.ok()) {
+    return absl::InvalidArgumentError(absl::StrFormat(
+        "\"audit_loggers[%d]\" %s", pos, result.status().message()));
```

Comment:
Maybe add something like this so it's clear the failing bit is from `AuditLoggerRegistry::ParseConfig`
"AuditLoggerRegistry::ParseConfig failed for audit_loggers[index] , <status message>"

Reply:
Similar to the reasoning above, `AuditLoggerRegistry` is not something people should know, they should only know which logger config has an error and it's the factory parser's job to return actionable errors, whether built-in or custom ones.

Reply:
Makes sense, the implementations of `ParseConfig` should be returning helpful errors that we pass up